solutions/ha-nva/layer-7-ingress/README.md

```diff
@@ -0,0 +1,106 @@
+
+## Deploy this template to Azure
+
+*Note: If you fork this repository, you will need to modify the link in [README.md](README.md) to point to your repo.  If you create a separate branch for testing, you will have to include a change to this link to point to your branch as well. You must include a URL-encoded link to the raw [azuredeploy.json](azuredeploy.json) file after `/uri/` in the link defined for the deployment button. You should also change the default value of `_artifactsLocation`.*
+
+#### Azure portal
+
+Use the following buttons to deploy the architecture using the Azure portal or to visualize with [armviz.io](https://armviz.io).
+
+[![Deploy To Azure](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.svg?sanitize=true)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fmspnp%2Fsamples%2Fmaster%2Fsolutions%2Fha-nva%2Flayer-7-ingress%2Fazuredeploy.json)
+[![Deploy To Azure US Gov](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazuregov.svg?sanitize=true)](https://portal.azure.us/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fmspnp%2Fsamples%2Fmaster%2Fsolutions%2Fha-nva%2Flayer-7-ingress%2Fazuredeploy.json)
+[![Visualize](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/visualizebutton.svg?sanitize=true)](http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2Fmspnp%2Fsamples%2Fmaster%2Fsolutions%2Fha-nva%2Flayer-7-ingress%2Fazuredeploy.json)
+
+#### Azure CLI
+
+Use the following command to create a resource group for the deployment.
+
+```bash
+az group create --name ha-nva-l7i --location eastus
+```
+
+Run the following command to deploy the Layer 7 Ingress example architecture.
+
+```bash
+az deployment group create --resource-group ha-nva-l7i \
+    --template-uri https://raw.githubusercontent.com/mspnp/samples/master/solutions/ha-nva/layer-7-ingress/azuredeploy.json
+```
+
+#### PowerShell
+
+Use the following command to create a resource group for the deployment.
+
+```powershell
+New-AzResourceGroup -Name ha-nva-l7i -Location eastus
+```
+
+Run the following command to deploy the Layer 7 Ingress example architecture.
+
+```powershell
+New-AzResourceGroupDeployment -ResourceGroupName ha-nva-l7i `
+    -TemplateUri https://raw.githubusercontent.com/mspnp/samples/master/solutions/ha-nva/layer-7-ingress/azuredeploy.json
+```
+
+#### Redeploying the architecture to the same Resource Group or Resource Group name
+
+You cannot redeploy this template to the same Resource Group without first deleting the NVA and web VM resources. The deployment script that generates the SSH key will generate a new key with each deployment and Azure does not allow redeployment of VMs that change the `linuxConfiguration.ssh.publicKeys` property.
```

Comment:
This is related to a conversation that we had over teams. What do you think about like a re-deployment condition that would then skip the VM resource when set?

May not matter, may add too much 'cruft' to the template.

Just bringing up for consideration.

Reply:
I've got this mostly re-deployable with one caveat noted in the README - it breaks if you chance the passphrase for the SSH key.